pkg/monitor/monitor/util.go

```diff
@@ -50,9 +50,52 @@ func buildTidbMonitorLabel(name string) map[string]string {
 	return label.NewMonitor().Instance(name).Monitor().Labels()
 }
 
+func getInitCommand(monitor *v1alpha1.TidbMonitor) []string {
+	c := `mkdir -p /data/prometheus
+chmod 777 /data/prometheus
+/usr/bin/init.sh`
+	if monitor.Spec.Grafana != nil {
+		c = `mkdir -p /data/prometheus /data/grafana
+chmod 777 /data/prometheus /data/grafana
+/usr/bin/init.sh`
+	}
+	command := []string{
+		"/bin/sh",
+		"-c",
+		c,
+	}
+	return command
+}
+
+func getGrafanaEnvs(monitor *v1alpha1.TidbMonitor, container *core.Container) {
```

Comment:
Please split this function into two functions and change the code logic as below:
```
if monitor.Spec.Grafana != nil {
  vol=append(vol, getGrafanaVols)
env=append(env, getGrafanaEnvs)
}
```

Reply:
fixed